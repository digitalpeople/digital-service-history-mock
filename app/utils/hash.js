import bcrypt from 'bcrypt';

const hash = password => bcrypt.hash(password, 10);
const compare = (actual, expected) => bcrypt.compare(actual, expected);

export { hash, compare };
