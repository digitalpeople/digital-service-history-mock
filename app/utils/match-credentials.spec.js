import matchCredentials from './match-credentials';
import { compare } from './hash';

jest.mock('./hash', () => ({
  compare: jest.fn(() => true),
}));

jest.mock('../../data/mock-credentials', () => ({
  mockUsername: 'test',
  mockPassword: 'mock-password',
}));

describe('matchCredentials', () => {
  it('should return true when credentials match to mocks', async () => {
    expect.assertions(1);

    const username = 'test';
    const password = 'mock-password';

    const actual = await matchCredentials(username, password);

    expect(actual).toMatchSnapshot();
  });

  it('should return false when credentials don\'t match to mocks', async () => {
    expect.assertions(1);

    const username = 'invalid-username';
    const password = 'mock-password';

    const actual = await matchCredentials(username, password);

    expect(actual).toMatchSnapshot();
  });

  it('should call compare with password parameter', async () => {
    expect.assertions(1);

    const username = 'test';
    const password = 'invalid-password';

    await matchCredentials(username, password);

    expect(compare.mock.calls[0]).toMatchSnapshot();
  });
});
