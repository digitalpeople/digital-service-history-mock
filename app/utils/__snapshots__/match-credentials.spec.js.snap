// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchCredentials should call compare with password parameter 1`] = `
Array [
  "invalid-password",
  "mock-password",
]
`;

exports[`matchCredentials should return false when credentials don't match to mocks 1`] = `false`;

exports[`matchCredentials should return true when credentials match to mocks 1`] = `true`;
